test/libsolidity/semanticTests/experimental/type_class.sol

```diff
@@ -1,5 +1,8 @@
 pragma experimental solidity;
 
+type word = __builtin("word");
+type bool = __builtin("bool");
```

Comment:
For type classes I'm going to introduce a very similar syntax:

```solidity
class integer = __builtin("integer");
```

Should be unambiguous in the parser.

Reply:
Actually it might be ambiguous for operators in the future, at least if we ever plan to introduce `=` or `:`. For example for `=` we'd get this:

```solidity
class = = __builtin("=");
```

should add some extra tokens to avoid this?
```solidity
class { = } = __builtin("=");
```

```solidity
class operator = = __builtin("=");
```

I suspect the answer is "no" or at least "not yet" since we're going to adjust the syntax anyway.

Reply:
I don't think we'll ever (or at least in a very long time) *define* ``=`` and ``:`` - they're too basic and special. In principle, it was also merely a shortcut to reuse the operator tokens as names for the classes - usually they're given proper names anyways (like ``Equal`` or ```Plus``) and the special syntax is defined separately (although we won't be targetting definable syntax in a long time, so the syntax would be an effect of the builtin marker)

Reply:
Marking as resolved since we discussed it in depth on the channel and I'm going to deal with type classes in a separate PR after all.